plugin/js/showpass-custom.js

```diff
@@ -285,14 +285,44 @@
 		const mutationObserver = new MutationObserver(function(mutations) {
 			mutations.forEach(function(mutation) {
 				if (mutation.target.className.includes('showpass-widget-body')) {
-					let gobj = window[window.GoogleAnalyticsObject];
-					let tracker, linker;
+
 					let iFrame = document.getElementById('showpass-widget');
+
+					// For analytics.js (UA)
+					// We use the linker provided by analytics.js to decorate the iframe src
+					let gobj = window[window.GoogleAnalyticsObject];
 					if (gobj) {
-						tracker = gobj.getAll()[0];
-						linker = new window.gaplugins.Linker(tracker);
+						let tracker = gobj.getAll()[0];
+						let linker = new window.gaplugins.Linker(tracker);
 						iFrame.src = linker.decorate(iFrame.src);
 					}
+
+					// For gtag.js (GA4)
+					// We use the gtag's get commands to get the client_id and session_id to decorate the iframe src
+					// This is additional to the analytics.js linker that should already be in place, AND the cross-domain tracking configured on the GA4 property itself
+					// which is for inbound/outbound clicks.
+					// @see https://support.google.com/analytics/answer/10071811?hl=en#zippy=%2Cmanual-setup
+					if (window.gtag && window.dataLayer) {
+						// Get the first available gtag config on the page. This config will be used
```

Comment:
if there are multiple does it always use the same client ID for all of them

Reply:
Yeah this would take the first configuration found and use it to pass onto the url similar to what our `analytics.js` code was doing before with
`tracker = gobj.getAll()[0]; `

 https://github.com/showpass/showpass-wordpress-plugin/blob/e01752ed2129e9f1bf353400dca8e4b02b0643ab/plugin/js/showpass-custom.js#L285-L298